airflow/www_rbac/templates/airflow/dag.html

```diff
@@ -280,208 +280,152 @@ <h4 class="modal-title" id="dagModalLabel">
-        "&dag_id=" + encodeURIComponent(dag_id) +
-        "&future=" + $('#btn_future').hasClass('active') +
-        "&past=" + $('#btn_past').hasClass('active') +
-        "&upstream=" + $('#btn_upstream').hasClass('active') +
-        "&downstream=" + $('#btn_downstream').hasClass('active') +
-        "&recursive=" + $('#btn_recursive').hasClass('active') +
-        "&execution_date=" + encodeURIComponent(execution_date) +
-        "&origin=" + encodeURIComponent(window.location);
-      window.location = url;
-    });
+$('#btn_task').click(() => {
+  const url = `{{ url_for('Airflow.task') }}?task_id=${encodeURIComponent(taskId)}&dag_id=${encodeURIComponent(dagId)}&execution_date=${encodeURIComponent(executionDate)}`;
+  window.location = url;
+});
 
-    $("#btn_dagrun_clear").click(function(){
-      url = "{{ url_for('Airflow.dagrun_clear') }}" +
-        "?task_id=" + encodeURIComponent(task_id) +
-        "&dag_id=" + encodeURIComponent(dag_id) +
-        "&execution_date=" + encodeURIComponent(execution_date) +
-        "&origin=" + encodeURIComponent(window.location);
-      window.location = url;
-    });
+$('#btn_ti').click(() => {
+  const url = `{{ url_for('TaskInstanceModelView.list') }}?flt1_dag_id_equals=${dagId}&_flt_3_task_id=${taskId}&_oc_TaskInstanceModelView=execution_date`;
+  window.location = url;
+});
 
-    $("#btn_failed").click(function(){
-      url = "{{ url_for('Airflow.failed') }}" +
-        "?task_id=" + encodeURIComponent(task_id) +
-        "&dag_id=" + encodeURIComponent(dag_id) +
-        "&upstream=" + $('#btn_failed_upstream').hasClass('active') +
-        "&downstream=" + $('#btn_failed_downstream').hasClass('active') +
-        "&future=" + $('#btn_failed_future').hasClass('active') +
-        "&past=" + $('#btn_failed_past').hasClass('active') +
-        "&execution_date=" + encodeURIComponent(execution_date) +
-        "&origin=" + encodeURIComponent(window.location);
+$('#btn_run').click(() => {
+  const url = `{{ url_for('Airflow.run') }}?task_id=${encodeURIComponent(taskId)}&dag_id=${encodeURIComponent(dagId)}&ignore_all_deps=${$('#btn_ignore_all_deps').hasClass('active')}&ignore_task_deps=${$('#btn_ignore_task_deps').hasClass('active')}&ignore_ti_state=${$('#btn_ignore_ti_state').hasClass('active')}&execution_date=${encodeURIComponent(executionDate)}&origin=${encodeURIComponent(window.location)}`;
+  window.location = url;
+});
 
-      window.location = url;
-    });
+$('#btn_clear').click(() => {
+  const url = `{{ url_for('Airflow.clear') }}?task_id=${encodeURIComponent(taskId)}&dag_id=${encodeURIComponent(dagId)}&future=${$('#btn_future').hasClass('active')}&past=${$('#btn_past').hasClass('active')}&upstream=${$('#btn_upstream').hasClass('active')}&downstream=${$('#btn_downstream').hasClass('active')}&recursive=${$('#btn_recursive').hasClass('active')}&execution_date=${encodeURIComponent(executionDate)}&origin=${encodeURIComponent(window.location)}`;
```

Comment:
could this be `let` ?

Reply:
Sure, it could be.  I preferred const over let anywhere the variable didn't get reassigned.  Is there an advantage to using let over const here?